api/producer/createDocumentReference/src/v1/handler.py

```diff
@@ -52,6 +53,17 @@ def _invalid_type(
     return source_document_pointer.type != target_document_pointer.type
 
 
+def _override_created_on(
+    data: PipelineData, document_pointer: DocumentPointer
+) -> DocumentPointer:
+    fhir_model: StrictDocumentReference = create_fhir_model_from_fhir_json(
+        fhir_json=data["body"]
+    )
+    if fhir_model.date is not None:
```

Comment:
Minor point but if we dont receive a date (eg this value is none) do we actually care about that or can we do as we do here and just carry on?

Reply:
It's potentially a redundant check tbh as setting the `fhir_model` will blow up if you don't provide it a date as part of the `create_fhir_model_from_fhir_json` method.